File: MicroTube.Client.Angular/src/app/data/DTO/CommentDTO.ts
import { DateTime } from "luxon";
import { LikeDislikeReactionDto } from "./LikeDislikeReactionDto";
import { LikeDislikeReactionsAggregationDto } from "./LikeDislikeReactionsAggregationDto";
export class CommentDto
{
  id: string;
  userId: string;
  userAlias: string | null;
  videoId: string;
  content: string;
  time: DateTime;
  reaction: LikeDislikeReactionDto | null;
  reactionsAggregation: LikeDislikeReactionsAggregationDto | null;
  deleted: boolean = false;
  constructor(raw: CommentRawDto)
  {
    this.id = raw.id;
    this.userId = raw.userId;
    this.userAlias = raw.userAlias;
    this.videoId = raw.videoId;
    this.content = raw.content;
    this.time = DateTime.fromISO(raw.time);
    this.reaction = raw.reaction;
    this.reactionsAggregation = raw.reactionsAggregation;
  }
  belongsToUser(userId: string | null)
  {
    if (userId == null)
    {
      return false;
    }
    return this.userId === userId;
  }
}
export interface CommentRawDto
{
  id: string;
  userId: string;
  userAlias: string | null;
  videoId: string;
  content: string;
  time: string;
  reaction: LikeDislikeReactionDto | null;
  reactionsAggregation: LikeDislikeReactionsAggregationDto | null;
}

File: MicroTube/Controllers/Comments/DTO/CommentDTO.cs
using MicroTube.Controllers.Reactions.Dto;

namespace MicroTube.Controllers.Comments.Dto
{
	public class CommentDto
	{
		public string Id { get; set; }
		public string? UserId { get; set; }
		public string? UserAlias { get; set; }
		public string? TargetId { get; set; }
		public string Content { get; set; }
		public DateTime Time { get; set; }
		public bool Edited { get; set; }
		public LikeDislikeReactionDto? Reaction { get; set; }
		public LikeDislikeReactionsAggregationDto? ReactionsAggregation {get;set;}
		public CommentDto(string id, string content, DateTime time, bool edited)
		{
			Id = id;
			Content = content;
			Time = time;
			Edited = edited;
		}
	}
}
